public/app/features/teams/state/reducers.ts

```diff
@@ -39,6 +40,6 @@ export const teamReducer = (state = initialTeamState, action: Action): TeamState
 };
 
 export default {
-  teams: teamsReducer,
-  team: teamReducer,
+  teams: resetStateForPath(teamsReducer, '/org/teams'),
+  team: resetStateForPath(teamReducer, '/org/teams'),
```

Comment:
Having path here & in router feels fragile, for granted very rare cases when urls change so should be fine. Why is team property being changed here when you move away from teams list?